### PR TITLE
Test /wallet/newAddress

### PR DESCRIPTION
**Pull request by Vavilen**
For [#640](https://github.com/skycoin/skycoin/issues/640)

### CODE REVIEW

File: src/gui/wallet_test.go [hunk 1]
```diff
@@ -15,6 +13,9 @@ import (
 
 	"github.com/stretchr/testify/require"
 
+	"bytes"
```
Comment (gz-c) [inline src/gui/wallet_test.go]: sort the imports properly

File: src/gui/wallet_test.go [hunk 2]
```diff
@@ -548,10 +555,191 @@ func TestWalletGet(t *testing.T) {
...
+				Id:  "foo",
+				Num: "1",
+			},
+			http.StatusBadRequest,
+			"400 Bad Request - gateway.NewAddresses error",
+			"foo",
+			1,
+			make([]cipher.Address, 0),
+			errors.New("gateway.NewAddresses error"),
+			Addresses{},
```
Comment (gz-c) [inline src/gui/wallet_test.go]: duplicate test case?

File: src/gui/wallet_test.go [hunk 3]
```diff
@@ -548,10 +555,191 @@ func TestWalletGet(t *testing.T) {
...
+				Id:  "foo",
+				Num: "1",
+			},
+			http.StatusOK,
+			"",
+			"foo",
+			1,
+			emptyAddrs,
+			nil,
+			responseEmptyAddresses,
```
Comment (gz-c) [inline src/gui/wallet_test.go]: what is this case testing?

File: src/gui/wallet_test.go (hunk 2 above)
Comment (Vavilen) [reply src/gui/wallet_test.go]: Not quite.

File: src/gui/wallet_test.go (hunk 3 above)
Comment (Vavilen) [reply src/gui/wallet_test.go]: This case shows that the result of gateway.NewAddresses can be an empty []cipher.Address, but the response status is 200.

File: src/gui/wallet_test.go (hunk 1 above)
Comment (Vavilen) [reply src/gui/wallet_test.go]: fixed

**Comment by gz-c** [comment]
merge conflict

File: src/gui/wallet_test.go (hunk 3 above)
Comment (gz-c) [reply src/gui/wallet_test.go]: ok, only when num is 0

File: src/gui/wallet_test.go (hunk 2 above)
Comment (gz-c) [reply src/gui/wallet_test.go]: it looks the same as the one about it

File: src/gui/wallet_test.go (hunk 3 above)
Comment (gz-c) [reply src/gui/wallet_test.go]: it should only return empty addrs if num is 0

File: src/gui/wallet_test.go (hunk 2 above)
Comment (gz-c) [reply src/gui/wallet_test.go]: *above it

Comment (Vavilen) [reply src/gui/wallet_test.go]: You're right. I will remove it.

File: src/gui/wallet_test.go (hunk 3 above)
Comment (Vavilen) [reply src/gui/wallet_test.go]: Changed 1 -> 0

**Pull request merged**